### PR TITLE
Add monotonically_increasing_id Spark function

### PR DESCRIPTION
**Pull request by zhli1142015**
Doc: https://spark.apache.org/docs/latest/api/sql/#monotonically_increasing_id

Impl: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala#L65C5-L65C19

### CODE REVIEW

**Comment by zhli1142015** [comment]
@PHILO-HE  and @rui-mo , could you help check? thanks.

File: velox/docs/functions/spark/misc.rst [hunk 1]
```diff
@@ -2,6 +2,13 @@
 Miscellaneous Functions
 ====================================
 
+.. spark:function:: monotonically_increasing_id() -> bigint
+
+    Returns monotonically increasing 64-bit integers. The generated Id is guaranteed to be monotonically increasing and unique, but not consecutive.
```
Comment (rui-mo) [inline velox/docs/functions/spark/misc.rst:7]: nit: Id -> ID?

File: velox/docs/functions/spark/misc.rst [hunk 2]
```diff
@@ -2,6 +2,13 @@
...
+    The current implementation puts the partition Id in the upper 31 bits, and the lower 33 bits represent the record number within each partition.
```
Comment (rui-mo) [inline velox/docs/functions/spark/misc.rst:8]: Id -> ID

File: velox/functions/sparksql/tests/MonotonicallyIncreasingIdTest.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+TEST_F(MonotonicallyIncreasingIdTest, basic) {
+  testMonotonicallyIncreasingId(0, 1, {0});
+  testMonotonicallyIncreasingId(2, 2, {17179869184, 17179869185});
+  testMonotonicallyIncreasingId(5, 3, {42949672960, 42949672961, 42949672962});
+  testMonotonicallyIncreasingId(
+      100, 4, {858993459200, 858993459201, 858993459202, 858993459203});
+}
+
+} // namespace
+} // namespace facebook::velox::functions::sparksql::test
```
Comment (rui-mo) [inline velox/functions/sparksql/tests/MonotonicallyIncreasingIdTest.cpp:48]: nit: needs one empty line.

File: velox/functions/sparksql/MonotonicallyIncreasingId.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions::sparksql {
+
+template <typename T>
+struct MonotonicallyIncreasingIdFunction {
+  static constexpr bool is_deterministic = false;
+
+  void initialize(const core::QueryConfig& config) {
+    count_ = (int64_t)config.sparkPartitionId() << 33;
```
Comment (rui-mo) [inline velox/functions/sparksql/MonotonicallyIncreasingId.h:27]: It looks like this functions fails if partition id not set. Maybe document this behavior as well.

**Review by rui-mo — COMMENTED** [review]
Thanks.

File: velox/docs/functions/spark/misc.rst (hunk 1 above)
Comment (zhli1142015) [reply velox/docs/functions/spark/misc.rst]: Updated, thanks.

File: velox/functions/sparksql/tests/MonotonicallyIncreasingIdTest.cpp (hunk 3 above)
Comment (zhli1142015) [reply velox/functions/sparksql/tests/MonotonicallyIncreasingIdTest.cpp]: Updated, thanks.

File: velox/functions/sparksql/MonotonicallyIncreasingId.h (hunk 4 above)
Comment (zhli1142015) [reply velox/functions/sparksql/MonotonicallyIncreasingId.h:27]: Added, thanks.

File: velox/docs/functions/spark/misc.rst [hunk 5]
```diff
@@ -2,6 +2,13 @@
 Miscellaneous Functions
 ====================================
 
+.. spark:function:: monotonically_increasing_id() -> bigint
+
+    Returns monotonically increasing 64-bit integers. The generated ID is guaranteed to be monotonically increasing and unique, but not consecutive.
```
Comment (PHILO-HE) [inline velox/docs/functions/spark/misc.rst:7]: Nit: please break such long line into two or more for being friendly to readers.

**Review by PHILO-HE — COMMENTED** [review]
Looks good! One small suggestion. Thanks!

File: velox/docs/functions/spark/misc.rst (hunk 5 above)
Comment (zhli1142015) [reply velox/docs/functions/spark/misc.rst]: Updated, thanks.

**Comment by zhli1142015** [comment]
Hello @mbasmanova , could you please help review? Thanks.

**Review by mbasmanova — APPROVED** [review]

**Comment by zhli1142015** [comment]
Yes, they are similar. One is implemented as a function and one is operator.
Thanks.

**Pull request closed without merging**